File: src/main/java/catmoe/fallencrystal/akanelimbo/command/SubCommand.java
package catmoe.fallencrystal.akanelimbo.command;

import java.util.List;
import java.util.Map;

import net.md_5.bungee.api.CommandSender;

public interface SubCommand {
    String getSubCommandId();

    void execute(CommandSender sender, String[] args);

    String getPermission();

    Map<Integer, List<String>> getTabCompleter();

    boolean allowedConsole();

    boolean strictSizeLimit();

    int strictSize();
}

File: src/main/java/catmoe/fallencrystal/akanelimbo/command/SubCommand.kt
package catmoe.fallencrystal.akanelimbo.command

import net.md_5.bungee.api.CommandSender

interface SubCommand {
    val subCommandId: String?
    fun execute(sender: CommandSender?, args: Array<String>)
    val permission: String?
    val tabCompleter: MutableMap<Int, List<String>>?
    fun allowedConsole(): Boolean
    fun strictSizeLimit(): Boolean
    fun strictSize(): Int
}
